File: Android/_Viper View_/Code/user_interface/VIPERView.java
//
//  __MODULENAME__View.java
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

package NAMESPACE.modules._VIPER_.user_interface;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import NAMESPACE.R;
import NAMESPACE.modules._VIPER_.contracts.VIPERContracts;

public class VIPERView extends AppCompatActivity implements VIPERContracts.View {

    public VIPERContracts.Presenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        this.presenter = new VIPERPresenter(this);
    }

    @Override
    protected void onDestroy() {
        this.presenter.unload();
        this.presenter = null;
        super.onDestroy();
    }
}

File: Kotlin/_VIPER View_/Code/user_interface/VIPERView.kt
//
//  __MODULENAME__View.kt
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

package NAMESPACE.modules._VIPER_.user_interface

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import NAMESPACE.modules._VIPER_.contracts.VIPERContracts

class VIPERView: AppCompatActivity(), VIPERContracts.View {
    private var presenter: VIPERContracts.Presenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        presenter = VIPERPresenter(this)
    }

    override fun onDestroy() {
        presenter?.unload()
        presenter = null
        super.onDestroy()
    }
}